src/arch/riscv/isa.cc

```diff
@@ -266,6 +265,16 @@ ISA::ISA(const Params &p) :
     _regClasses.push_back(&ccRegClass);
     _regClasses.push_back(&miscRegClass);
 
+    system = dynamic_cast<RiscvSystem *>(p.system);
+
+    if (FullSystem && system){
```

Comment:
I still don't understand the need for `system` here. I think it should be removed.

Another way to look at this... these parameters `VLEN` and `ELEN` are specific to a *hart*. In gem5, the ISA object is the place where we specify the hart parameters.